addon/resolver.js

```diff
@@ -65,7 +78,7 @@ export default EmberResolver.extend({
     let { resolvingType, resolvingPath } = { resolvingType: fullNamePartsArray[0], resolvingPath: fullNamePartsArray[1] };
     let resolvingPathParts = resolvingPath.split('/');
 
-    if (!this._resolveResourceWithoutDeviceTypeDetection(fullName)) {
+    if (this._resolveTypeWithDeviceTypeDetection(resolvingType) && !this._resolveResourceWithoutDeviceTypeDetection(fullName)) {
       let pathPrefixes = device.pathPrefixes(true);
```

Comment:
To increase performanse of 'resolve' method, we can also reduce count of path prefixes.
For example for tablet on android following paths will be checked (in same order):
   // Paths related to platform & type & orientation.
   - templates/android-tablet-landscape/my-form.hbs
   - templates/android-tablet-portrait/my-form.hbs
   
   // Paths related for platform & type.
   - templates/android-tablet/my-form.hbs
   
   // Path related to type & orientation.
   - templates/tablet-landscape/my-form.hbs
   - templates/tablet-portrait/my-form.hbs

   // Path related to type only.   
   - templates/tablet/my-form.hbs
   
   // Common path for non desktop devices.
   - templates/mobile/my-form.hbs
   
   // Default path.
   - templates/my-form.nbs

Some seting can be added to resolver & to device.pathPrefixes method, if setting is defined reduced prefixes count will be returned by device.pathPrefixes.